File: samplejava/src/main/java/com/kazakago/factolize/samplejava/MainActivity.java
package com.kazakago.factolize.samplejava;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v4.app.FragmentManager;
import android.support.v4.app.FragmentTransaction;
import android.support.v7.app.AppCompatActivity;

import com.kazakago.factolize.Factory;

@Factory
public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        MainActivityFactory.injectArgument(this);

        if (savedInstanceState == null) {
            replaceMainFragment();
        }
    }

    private void replaceMainFragment() {
        MainFragment fragment = MainFragmentFactory.createInstance();

        FragmentManager fragmentManager = getSupportFragmentManager();
        FragmentTransaction fragmentTransaction = fragmentManager.beginTransaction();
        fragmentTransaction.replace(R.id.fragmentContainer, fragment);
        fragmentTransaction.commit();
    }

}

File: samplekotlin/src/main/java/com/kazakago/factolize/samplekotlin/MainActivity.kt
package com.kazakago.factolize.samplekotlin

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.kazakago.factolize.Factory

@Factory
class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        MainActivityFactory.injectArgument(this)

        if (savedInstanceState == null) {
            replaceMainFragment()
        }
    }

    private fun replaceMainFragment() {
        val fragment = MainFragmentFactory.createInstance()

        val fragmentManager = supportFragmentManager
        val fragmentTransaction = fragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.fragmentContainer, fragment)
        fragmentTransaction.commit()
    }

}
